eth/tracers/firehose_concurrency.go

```diff
@@ -0,0 +1,105 @@
+package tracers
+
+import (
+	pbeth "github.com/streamingfast/firehose-ethereum/types/pb/sf/ethereum/type/v2"
+	"sync"
+)
+
+type blockPrintJob struct {
+	block    *pbeth.Block
+	finality *FinalityStatus
+}
+
+type outputJob struct {
+	blockNum uint64
+	data     []byte
+}
+
+type BlockFlushQueue struct {
+	bufferSize int
+
+	startSignal    chan uint64
+	jobQueue       chan *blockPrintJob
+	outputQueue    chan *outputJob
+	printBlockFunc func(block *pbeth.Block, finality *FinalityStatus)
+	outputFunc     func([]byte)
+
+	jobWG     sync.WaitGroup
+	outputWG  sync.WaitGroup
+	closeOnce sync.Once
+}
+
+func NewBlockFlushQueue(concurrency int, bufferSize int, printBlockFunc func(*pbeth.Block, *FinalityStatus), outputFunc func([]byte)) *BlockFlushQueue {
+	if concurrency <= 0 {
+		panic("BlockFlushQueue requires concurrency > 0")
+	}
+
+	q := &BlockFlushQueue{
+		startSignal:    make(chan uint64, 1),
+		jobQueue:       make(chan *blockPrintJob, bufferSize),
+		outputQueue:    make(chan *outputJob, bufferSize),
+		outputFunc:     outputFunc,
+		bufferSize:     bufferSize,
+		printBlockFunc: printBlockFunc,
+	}
+
+	for i := 0; i < concurrency; i++ {
+		q.jobWG.Add(1)
+		go q.worker()
+	}
```

Comment:
No start on constructor, not a good pattern. Have a dedicated method instead so the lifecycle is clearer.